### PR TITLE
[plugin.video.themoviedb.helper] 5.2.25

### PR DESCRIPTION
**Pull request by jurialmunkey**
### Description

Add and update TMDbHelper for Nexus and higher.

**NOTE**
Requires script.module.jurialmunkey to be updated in this PR first to meet dep requirements: https://github.com/xbmc/repo-scripts/pull/2624

Acts as a helper plugin to generate widgets and retrieve details from online APIs for skinners. See wiki for full feature list
https://github.com/jurialmunkey/plugin.video.themoviedb.helper/wiki

Note an older version exists on Matrix repo. However, I'm submitting new for Nexus as the current version now uses the new infotag setters and so now only works on Nexus or higher.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

**Comment by jurialmunkey** [comment]
> packaging.version.InvalidVersion: Invalid version: '5.2.25~nexus'

I don't understand this error? This is a valid version afaik.
Is the tilde `~` not allowed in the repo?

EDIT: I've removed the `~nexus` suffix so the checker doesn't complain.

**Comment by jurialmunkey** [comment]
I've fixed all the errors. Checker is still complaining because of the dependency module that needs to go in first but otherwise should be clean now.

**Comment by Hitcher** [comment]
Does the addon checker need running again for this (and script.skinvariables) now that script.module.jurialmunkey v0.1.18 has been committed?

**Comment by jurialmunkey** [comment]
> Does the addon checker need running again for this (and script.skinvariables) now that script.module.jurialmunkey v0.1.18 has been committed?

Yeah, it looks like Bas tried yesterday but it hadn't quite hit the repo yet so still failed. I just retriggered now by force pushing a blank commit amend and looks like green light now.

**Comment by kris06** [comment]
major problem on search, 'the accents are not supported' on Fuse

Example 
(en francais) présumé innocent is invalid  
presumed innocent is valid research

**Comment by Hitcher** [comment]
@kris06 this isn't the correct place for issues. Please go here instead: https://github.com/jurialmunkey/plugin.video.themoviedb.helper/issues

**Comment by Hitcher** [comment]
Is there any particular reason this is being overlooked?
Anything we can do to help get it merged?

**Comment by basrieter** [comment]
> Is there any particular reason this is being overlooked? Anything we can do to help get it merged?

It's not being overlooked, but the PR is huge, so it takes up a lot of time to review. And thus is not a PR that we can quickly check and merge when we have some minutes lefts. 

I will start a review today and see if I can fine a larger slot of time. But for the future: if you keep the PR's a bit smaller, things do go faster: 2 smaller PR's (after one another) will go faster than one huge one.

File: plugin.video.themoviedb.helper/depends.txt
Comment (basrieter) [inline plugin.video.themoviedb.helper/depends.txt:1]: Is this file needed for the Kodi repo, seems like a dev file? Please remove.

File: plugin.video.themoviedb.helper/icon.xcf
Comment (basrieter) [inline plugin.video.themoviedb.helper/icon.xcf:1]: Is this file needed for the Kodi repo? I don't think that Kodi handles `.xcf`-files. Please remove.

File: plugin.video.themoviedb.helper/addon.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  <import addon="script.module.beautifulsoup4" version="4.9.3" />
+</requires>
+<extension point="xbmc.python.pluginsource" library="resources/plugin.py">
+  <provides>video</provides>
+</extension>
+<extension point="xbmc.python.script" library="resources/script.py" />
+<extension point="xbmc.service" library="resources/service.py" />
+<extension point="xbmc.python.module" library="resources/modules"/>
+<extension point="xbmc.addon.metadata">
+  <reuselanguageinvoker>false</reuselanguageinvoker>
```
Comment (basrieter) [inline plugin.video.themoviedb.helper/addon.xml:22]: Any reason why `reuselanguageinvoker` is disabled? This will require a new Python invoker with each run, instead of re-using the existing one with subsequent plugin requests. Re-using saves a lot of CPU and is faster.

**Review by basrieter — COMMENTED** [review]
Did a first check of non-Python stuff.

**Comment by Hitcher** [comment]
> It's not being overlooked, but the PR is huge, so it takes up a lot of time to review. And thus is not a PR that we can quickly check and merge when we have some minutes lefts.

Thanks for the update.

> I will start a review today and see if I can fine a larger slot of time.

Thanks, I really appreciate it as this is the last dependency needed before I can submit my skin to the Omega repo.

File: plugin.video.themoviedb.helper/addon.xml (hunk 1 above)
Comment (jurialmunkey) [reply plugin.video.themoviedb.helper/addon.xml:22]: Because of this long standing bug which causes Kodi to hang and/or crash when reuselanguageinvoker plugins are used as widgets: https://github.com/xbmc/xbmc/issues/21653

File: plugin.video.themoviedb.helper/icon.xcf
Comment (jurialmunkey) [reply plugin.video.themoviedb.helper/icon.xcf:1]: Done. Forgot that these two files were in my github.

File: plugin.video.themoviedb.helper/depends.txt
Comment (jurialmunkey) [reply plugin.video.themoviedb.helper/depends.txt:1]: Done

**Comment by jurialmunkey** [comment]
> I will start a review today and see if I can fine a larger slot of time. But for the future: if you keep the PR's a bit smaller, things do go faster: 2 smaller PR's (after one another) will go faster than one huge one.

Yeah this is my fault. I'm really sorry about getting so tardy on submitting updates as I know this is a massive pita to review.

BTW - the plugin is on the Matrix repo but I needed to make a new submission against Nexus as the info tag changes means it no longer works on Matrix -- it's still huge but if you compare against that version might give some indication of where there are trivial vs. significant changes. A lot is refactoring pushing things around as I try to unpick some of my poorer choices from when I was beginning to learn python (my eventual goal was to modularise things to make it all a bit more manageable but release cycle beat time/motivation available - so unfortunately it's still a mammoth).

If it helps at all here's the compare from my repo for the tag of the version in Matrix repo (v4.10.14) and this PR:
https://github.com/jurialmunkey/plugin.video.themoviedb.helper/compare/v4.10.14..v5.2.25

**Review by basrieter — APPROVED** [review]

**Comment by basrieter** [comment]
So it took me quite some time but I got through it. Hopefully the future PR's are smaller :)

**Comment by Hitcher** [comment]
That's awesome news. Many thanks basieter, it's really appreciated.

**Comment by jurialmunkey** [comment]
> So it took me quite some time but I got through it. Hopefully the future PR's are smaller :)

Yes definitely will be I promise! Really appreciate you taking the time to review.
